### DISCUSSION TITLE
Editing the demo without a login would be nice

### DISCUSSION THREAD

**Discussion by guettli** [django-wiki/django-wiki#1263] [Ideas]
It would be nice, if you could edit some pages in the demo without creating an account.

https://demo.django-wiki.org/

**Comment by benjaoming** [comment]
Oh yeah sorry about that... this has needed proper addressing for a long time! Basically, discovering and understanding django-wiki lacks some guidance.

For a better experience, we could:

* [x] Add a basic demo account
  * Use `admin:admin` for the user and password
* [x] Create a setting for demo mode in `wiki.settings`
  * Done via #1267 
* [x] Improve demo contents and turn them into fixtures (having that sqlite db in git is not so cool)
  * Done via #1260
* [x] Instruction message box for demo mode w/ login instructions
  * Done via #1267

**Reply by oscarmcm** [reply]
@benjaoming Is there a way I can get more information about the server setup?
- does it runs in a container or so?
- is gunicorn and supervisor used?

So I can bootstrap a simple shell script, that then we'll added this to a cronjob for every 10 mins and will do (from top of my head):
- Stop App Server
- Delete database (sqlite will be enough here)
- Create a new database
- Load fixtures
- Start App Server

**Reply by benjaoming** [reply]
It's running on a VPS, no containers. It uses uWSGI and can reload automatically.

The points above are already done automatically in a cronjob every 2 hours. Except for loading fixtures, but I can definitely add that :)

It also deletes media files so the attachment feature is reset. And it installs all latest dependencies and adds the secret environment.

I think every 10 minutes can make it disturbing/stressful to test the demo. But this interval would be nice to have as part of a message box/banner: "This is a demo, every 2 hours all data is removed. Logins available: admin/admin and user/user"

**Reply by oscarmcm** [reply]
That's what I'm talking about!! Letssss gooooo 🚀 

I'll open a new MR for the banner, and agree 2hrs will be good.

**Reply by oscarmcm** [reply]
I'm also thinking that we should remove the password for the `djangowiki` user every time the app is fresh loaded.

I believe the password is secure, but it's better to remove it so we don't have to deal with "hack" attempts.

**Reply by benjaoming** [reply]
That's a good idea.

The script for flushing the demo has been updated now to use the fixtures instead :+1:

**Comment by oscarmcm** [comment]
@benjaoming I believe we've already completed all the necessary configuration in order to make the demo more usable for visitors.

**Reply by benjaoming** [reply]
Yes agreed, I think just some regression in the default setup... that admin should have admin rights and if we want to add a normal user, it should own an article or something.

But otherwise, looks good :+1:

**Reply by benjaoming** [reply]
Oh I think the message will be spawned twice upon a redirect (guessing because it passes through the redirect w/o getting displayed)

![image](https://user-images.githubusercontent.com/374612/218506620-40e3ea9a-9cf2-45a4-b285-654c3ecb265d.png)

Maybe we should only add it once per session..
